src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py

```diff
@@ -0,0 +1,316 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+# noqa: E501
+import asyncio
+import logging
+from typing import Any, Callable, Dict, List
+
+from tqdm import tqdm
+
+from promptflow._sdk._telemetry import ActivityType, monitor_operation
+
+from ._conversation import CallbackConversationBot, ConversationBot, ConversationRole
+from ._conversation._conversation import simulate_conversation
+from ._model_tools import (
+    AdversarialTemplateHandler,
+    AsyncHTTPClientWithRetry,
+    ManagedIdentityAPITokenManager,
+    ProxyChatCompletionsModel,
+    RAIClient,
+    TokenScope,
+)
+from ._utils import JsonLineList
+
+logger = logging.getLogger(__name__)
+
+
+class AdversarialSimulator:
+    @monitor_operation(activity_name="adversarial.simulator.init", activity_type=ActivityType.PUBLICAPI)

```

Comment:
telemetry on call should be enough?  no need to add for __init?

Reply:
removed it.